examples/todo-app/web/src/App.tsx

```diff
@@ -0,0 +1,117 @@
+import { FaCirclePlus } from "react-icons/fa6";
+import { SlDrawer } from "react-icons/sl";
+import Button from "./components/Button";
+import Header from "./components/Header";
+
+import { useState } from "react";
+import Options from "./components/Options";
+import CreateTodoModal from "./components/todos/CreateTodoModal";
+import TodoCard from "./components/todos/TodoCard";
+import { useTodo } from "./context/TodoContext";
+import { classNames } from "./utils";
+
+function App() {
+  const { todos } = useTodo();
+
+  const [createTodoModal, setCreateTodoModal] = useState<boolean>(false);
+  const [selectedOption, setSelectedOption] = useState<string>("all");
+
+  const closeCreateTodoModal = () => {
+    setCreateTodoModal(false);
+  };
+  const openCreateTodoModal = () => {
+    setCreateTodoModal(true);
+  };
+
+  const optionsList = [
+    {
+      title: "All Todos",
+      slug: "all",
+      count: todos?.length,
+    },
+    {
+      title: "Pending",
+      slug: "pending",
+      count: todos?.filter((todo) => !todo.isComplete).length,
+    },
+    {
+      title: "Completed",
+      slug: "completed",
+      count: todos?.filter((todo) => todo.isComplete).length,
+    },
+  ];
+
+  return (
+    <>
+      <Header openTodoModal={openCreateTodoModal} />
+
+      {createTodoModal && <CreateTodoModal onClose={closeCreateTodoModal} />}
+
+      {todos && todos.length > 0 ? (
+        <div className={classNames("flex flex-col items-center p-3 ")}>
+          <div className=" w-full px-3 mt-10">
+            <Button
+              onClick={openCreateTodoModal}
+              severity="primary"
+              className="mx-auto w-full max-w-[500px]"
+            >
+              <FaCirclePlus />
+              Create a new task
+            </Button>
+          </div>
+
+          <div className="w-full flex justify-around max-w-[600px] gap-4 my-6">
+            {optionsList.map((option) => (
+              <Options
+                isActive={selectedOption === option.slug}
+                key={option.slug}
+                title={option.title}
+                count={option.count}
+                onClick={() => {
+                  setSelectedOption(option.slug);
+                }}
+              />
+            ))}
+          </div>
+
+          <div className=" w-full max-w-[800px] flex flex-col overflow-auto h-[calc(100vh-300px)] border-t-2">
+            {todos
+              .filter((todo) =>
```

Comment:
The backend has `complete` query parameter which gives you `pending`, `completed` todos. If you don't pass `complete` it will send you `all` the todos.

We can call get all todos api every time user refreshes the page. No need to store it in `LocalStorage`. And call the get all todos api with `complete` query value based on selected tab. 

reference: https://api.freeapi.app/#/%F0%9F%93%9D%20Todo%20list/getAllTodos

Reply:
So should I remove all the localstorage part

Reply:
I have pushed the updated code, do let me know if there are more changes to be made